### PR TITLE
[INLONG-8845][Manager] Support Tencent Cloud Log Service data flow

### LINKED ISSUE #8845: [Feature][Manager] Support Tencent Cloud Log Service data flow
### Description

support tencent cloud Log Service data flow

Tencent Cloud Log Service detailed :
Tencent Cloud Log Service (CLS) is a one-stop solution that offers real-time log collection, storage, search, analysis, consuming and shipping, enabling businesses to meet their operational, security, regulatory and analytical needs
[https://cloud.tencent.com/document/product/614](url)
### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by castorqin**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8845 

### Motivation

Support Tencent Cloud Log Service data flow

### Modifications
The details can be found at [#8845 ](https://github.com/apache/inlong/issues/8845).

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@ApiModel("Tencent cloud log service data node info")
+public class TencentClsDataNodeDTO {
+
+    /**
+     *Tencent cloud log service master account
+     */
+    private String mainAccountID;
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java:47]: mainAccountId

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     *Tencent cloud log service subAccount
+     */
+    private String subAccountID;
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java:52]: subAccountId

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * Tencent cloud log service region
+     */
+    private String region;
+
+    /**
+     * Tencent cloud log service  set id
+     */
+    private String logSetID;
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java:87]: logSetId

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeInfo.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeDefine(value = DataNodeType.ELASTICSEARCH)
+@ApiModel("Tencent cloud log service data node info")
+public class TencentClsDataNodeInfo extends DataNodeInfo {
+
+    /**
+     *Tencent cloud log service master account
+     */
+    private String mainAccountID;
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeInfo.java:48]: Ditto.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeRequest.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeDefine(value = DataNodeType.CLS)
+@ApiModel("Tencent cloud log service data node request")
+public class TencentClsDataNodeRequest extends DataNodeRequest {
+
+    /**
+     *Tencent cloud log service master account
+     */
+    private String mainAccountID;
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeRequest.java:42]: Please add ` @ApiModelProperty`

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        try {
+            Credential cred = new Credential(tencentClsDataNode.getManageSecretId(),
+                    tencentClsDataNode.getManageSecretId());
+            HttpProfile httpProfile = new HttpProfile();
+            httpProfile.setEndpoint(tencentClsDataNode.getEndpoint());
+            ClientProfile clientProfile = new ClientProfile();
+            clientProfile.setHttpProfile(httpProfile);
+            ClsClient client = new ClsClient(cred, tencentClsDataNode.getRegion(), clientProfile);
+            CreateTopicRequest req = new CreateTopicRequest();
+            String[] allTags = tencentClsSinkDTO.getTag().split("\\|");
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java:92]: Please add to  `InlongConstanst`

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+            CreateTopicResponse resp = client.CreateTopic(req);
+            LOG.info("create cls topic {} success ,topicId {}", tencentClsSinkDTO.getTopicName(), resp.getTopicId());
+            tencentClsSinkDTO.setTopicID(resp.getTopicId());
+            sinkInfo.setExtParams(JsonUtils.toJsonString(tencentClsSinkDTO));
+            StreamSinkEntity streamSinkEntity = new StreamSinkEntity();
+            CommonBeanUtils.copyProperties(sinkInfo, streamSinkEntity, true);
+            streamSinkEntityMapper.updateByIdSelective(streamSinkEntity);
+            String info = "success to create cls resource";
+            sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_SUCCESSFUL.getCode(), info);
+            LOG.info("update cls sink info status {} success ,topicId {}", tencentClsSinkDTO.getTopicName(),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java:106]: topicName

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+            sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_FAILED.getCode(), errMsg);
+            throw new WorkflowException(errMsg);
+        }
+    }
+
+    private Tag[] convertTags(String[] allTags) {
+        Tag[] tags = new Tag[allTags.length];
+        for (int i = 0; i < allTags.length; i++) {
+            String tag = allTags[i];
+            String[] keyAndValueOfTag = tag.split(":");
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java:120]: Please use `InlongConstants.COLON`

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeRequest.java (hunk 5 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeRequest.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java (hunk 7 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java (hunk 8 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeInfo.java (hunk 4 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeInfo.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java (hunk 3 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java (hunk 2 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java (hunk 1 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/TencentClsDataNodeDTO.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java (hunk 6 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/TencentClsResourceOperator.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * cloud log service data node info
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java:35]: Cloud

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+ */
+@Data
+@SuperBuilder
+@AllArgsConstructor
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeDefine(value = DataNodeType.ELASTICSEARCH)
+@ApiModel("cloud log service data node info")
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java:43]: Cloud

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+import org.apache.inlong.manager.pojo.node.DataNodeRequest;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+/**
+ * cloud log service data node request
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java:31]: Ditto.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+ */
+@Data
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeDefine(value = DataNodeType.CLS)
+@ApiModel("cloud log service data node request")
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java:37]: Ditto.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+    private void createTopicID(SinkInfo sinkInfo) {
+        ClsDataNodeDTO clsDataNode = getClsDataNode(sinkInfo);
+        ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
+        try {
+            ClsClient client = getClsClient(clsDataNode);
+            CreateTopicRequest req = new CreateTopicRequest();
+            String[] allTags = clsSinkDTO.getTag().split(InlongConstants.CENTER_LINE);
+            Tag[] tags = convertTags(allTags);
+            req.setTags(tags);
+            req.setLogsetId(clsDataNode.getLogSetId());
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:93]: Wrap this code into a method.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+            throw new BusinessException(errMsg);
+        }
+    }
+
+    private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
+        Credential cred = new Credential(clsDataNode.getManageSecretId(),
+                clsDataNode.getManageSecretId());
+        HttpProfile httpProfile = new HttpProfile();
+        httpProfile.setEndpoint(clsDataNode.getEndpoint());
+        ClientProfile clientProfile = new ClientProfile();
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:120]: Add Blank Line.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+        clientProfile.setHttpProfile(httpProfile);
+        return new ClsClient(cred, clsDataNode.getRegion(), clientProfile);
+    }
+
+    private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
+        ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
+        if (Strings.isEmpty(clsSinkDTO.getTokenizer())) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:127]: StringUtils.isNotBlank

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/cls/ClsSinkOperator.java [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+        DataNodeEntity dataNodeEntity = dataNodeEntityMapper.selectByUniqueKey(streamSink.getDataNodeName(),
+                DataNodeType.CLS);
+        ClsDataNodeDTO clsDataNodeDTO = JsonUtils.parseObject(dataNodeEntity.getExtParams(),
+                ClsDataNodeDTO.class);
+        params.put(SECRET_ID, clsDataNodeDTO.getSendSecretId());
+        params.put(SECRET_KEY, clsDataNodeDTO.getSendSecretKey());
+        params.put(END_POINT, clsDataNodeDTO.getEndpoint());
+        StringBuilder fieldNames = new StringBuilder();
+        for (String field : fields) {
+            fieldNames.append(field).append(" ");
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/cls/ClsSinkOperator.java:125]: Use `InlongConstants.BLANK`

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java (hunk 9 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java (hunk 10 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeInfo.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java (hunk 11 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 13 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 15 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/cls/ClsSinkOperator.java (hunk 16 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/cls/ClsSinkOperator.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java (hunk 12 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/node/cls/ClsDataNodeRequest.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 14 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:123]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/cls/ClsSinkDTO.java [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+    private String tag;
+
+    @ApiModelProperty("Cloud log service index tokenizer")
+    private String tokenizer;
+
+    /**
+     * Get the dto instance from the request
+     */
+    public static ClsSinkDTO getFromRequest(ClsSinkRequest request, String extParams) {
+        ClsSinkDTO dto =
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/cls/ClsSinkDTO.java:62]:
```suggestion
        ClsSinkDTO dto = StringUtils.isNotBlank(extParams) 
                ? ClsSinkDTO.getFromJson(extParams) 
                : new ClsSinkDTO();
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    protected void setTargetEntity(DataNodeRequest request, DataNodeEntity targetEntity) {
+        ClsDataNodeRequest esRequest = (ClsDataNodeRequest) request;
+        CommonBeanUtils.copyProperties(esRequest, targetEntity, true);
+        try {
+            ClsDataNodeDTO dto =
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java:58]: ClsDataNodeDTO dto = ClsDataNodeDTO.getFromRequest( clsDataNodeRequest, targetEntity.getExtParams());

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+public class ClsDataNodeOperator extends AbstractDataNodeOperator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClsDataNodeOperator.class);
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    protected void setTargetEntity(DataNodeRequest request, DataNodeEntity targetEntity) {
+        ClsDataNodeRequest esRequest = (ClsDataNodeRequest) request;
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java:55]: clsDataNodeRequest

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 20]
```diff
@@ -0,0 +1,172 @@
...
+            ClsClient client = getClsClient(clsDataNode);
+            CreateTopicRequest req = getCreateTopicRequest(clsDataNode, clsSinkDTO);
+            CreateTopicResponse resp = client.CreateTopic(req);
+            LOG.info("create cls topic {} success ,topicId {}", clsSinkDTO.getTopicName(), resp.getTopicId());
+            clsSinkDTO.setTopicId(resp.getTopicId());
+            sinkInfo.setExtParams(JsonUtils.toJsonString(clsSinkDTO));
+            StreamSinkEntity streamSinkEntity = new StreamSinkEntity();
+            CommonBeanUtils.copyProperties(sinkInfo, streamSinkEntity, true);
+            streamSinkEntityMapper.updateByIdSelective(streamSinkEntity);
+            this.createTopicIndex(sinkInfo);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:96]:
```suggestion
            this.createTopicIndex(sinkInfo);
             CommonBeanUtils.copyProperties(sinkInfo, streamSinkEntity, true);
            streamSinkEntityMapper.updateByIdSelective(streamSinkEntity);
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+            return;
+        }
+        ClsDataNodeDTO clsDataNode = getClsDataNode(sinkInfo);
+        ClsClient clsClient = getClsClient(clsDataNode);
+        CreateIndexRequest req = new CreateIndexRequest();
+        RuleInfo ruleInfo = new RuleInfo();
+        FullTextInfo fullTextInfo = new FullTextInfo();
+        fullTextInfo.setTokenizer(clsSinkDTO.getTokenizer());
+        ruleInfo.setFullText(fullTextInfo);
+        req.setRule(ruleInfo);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:142]:
```suggestion
        RuleInfo ruleInfo = new RuleInfo();
        FullTextInfo fullTextInfo = new FullTextInfo();
        fullTextInfo.setTokenizer(clsSinkDTO.getTokenizer());
        ruleInfo.setFullText(fullTextInfo);
        
        CreateIndexRequest req = new CreateIndexRequest();
        req.setRule(ruleInfo);
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 20 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:96]: create an index,  must first fill in the topicId

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 21 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java (hunk 19 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java (hunk 18 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/cls/ClsDataNodeOperator.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/cls/ClsSinkDTO.java (hunk 17 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/cls/ClsSinkDTO.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 22]
```diff
@@ -0,0 +1,175 @@
...
+            LOG.warn("sink resource [" + sinkInfo.getId() + "] already success, skip to create");
+            return;
+        } else if (InlongConstants.DISABLE_CREATE_RESOURCE.equals(sinkInfo.getEnableCreateResource())) {
+            LOG.warn("create resource was disabled, skip to create for [" + sinkInfo.getId() + "]");
+            return;
+        }
+        this.createTopicID(sinkInfo);
+    }
+
+    private void createTopicID(SinkInfo sinkInfo) {
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:84]: Add some comments plz.

Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:87]: done

**Review by vernedeng — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**